reinhard/util/dependencies.py

```diff
@@ -49,19 +49,37 @@ def __call__(self) -> aiohttp.ClientSession:
         return self._session
 
 
-class PaginatorPoolDependency:
-    __slots__ = ("_paginator",)
+class ReactionClientDependency:
+    __slots__ = ("_client",)
 
     def __init__(self) -> None:
-        self._paginator: paginaton.PaginatorPool | None = None
+        self._client: yuyo.ReactionClient | None = None
 
     def __call__(
         self,
         rest_client: traits.RESTAware = injecting.injected(type=traits.RESTAware),
         event_client: traits.EventManagerAware = injecting.injected(type=traits.EventManagerAware),
-    ) -> paginaton.PaginatorPool:
-        if not self._paginator or self._paginator.is_closed:
-            self._paginator = paginaton.PaginatorPool(rest_client, event_client)
-            asyncio.create_task(self._paginator.open())
+    ) -> yuyo.ReactionClient:
+        if not self._client or self._client.is_closed:
+            self._client = yuyo.ReactionClient(rest=rest_client.rest, event_manager=event_client.event_manager)
```

Comment:
Once generic dependency injection is implemented in Tanjun and includes all client callbacks, an "open" method can be added to this class for actually starting it instead of checking if we need to start on every call which is racey